### ISSUE TITLE
[java] UnusedPrivateField cannot override ignored annotations property

### ISSUE THREAD

**Issue by martin-tarjanyi** [pmd/pmd#2876]
**Description:**

There is this rule: https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatefield

By default this rule will ignore any classes which annotated with any of the mentioned annotations. Now, I would like to override this and I want PMD to ignore these classes only if they are marked with lombok `@Getter` or `@Data` but do not ignore when they are marked with `@RequiredArgsConstructor` for example:

So I did this in my rule set:
```xml
    <rule ref="category/java/bestpractices.xml/UnusedPrivateField">
        <properties>
            <property name="ignoredAnnotations" value="lombok.Getter|lombok.Data" />
        </properties>
    </rule>
```

However, this doesn't seem to work as the rule still doesn't trigger for a class which is only annotated with `@RequiredArgsConstructor` but not with `@Data` or `@Getter`. My impression is that when I override this property in my rule set it doesn't actually override the default list, instead it just extends it so `@RequiredArgsConstructor` is still part of the value. Is that right? Is it possible to completely override this property?

**Comment by adangel** [comment]
At first glance, I'd say your understanding/expectation is correct: The property value should be completely overridden.

One possible explanation could be, if you have this rule multiple times in your ruleset. Then the last configuration will be used (ignoring the previous ones).

E.g. if you first configure this rule as described above, but after that import the complete bestpractices category (e.g. via `<rule ref="category/java/bestpractices.xml"/>`), then the latter reference would pull in UnusedPrivateField with the default configuration, overwriting your custom config.

Can you share you complete ruleset.xml file you are using?

**Comment by martin-tarjanyi** [comment]
Created a sample repo: https://github.com/martin-tarjanyi/pmd-bug-repro

Ruleset: https://github.com/martin-tarjanyi/pmd-bug-repro/blob/master/pmd_ruleset.xml
Violation: https://github.com/martin-tarjanyi/pmd-bug-repro/blob/master/src/main/java/org/example/App.java#L11-L14

**Comment by oowekyala** [comment]
Thanks for making an MWE. This is a false negative, the rule actually whitelists classes that have *any* lombok annotation, regardless of the property:

https://github.com/pmd/pmd/blob/9f701ee460fcfdc781ebc5d96c46fbe5fbdac79f/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateFieldRule.java#L45

then

https://github.com/pmd/pmd/blob/9f701ee460fcfdc781ebc5d96c46fbe5fbdac79f/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateFieldRule.java#L52-L55
